be/src/common/configbase.cpp

```diff
@@ -308,6 +318,35 @@ bool init(const char* filename, bool fillconfmap) {
         SET_FIELD(it.second, std::vector<std::string>, fillconfmap);
     }
 
+    // if user hasn't set flush_thread_num_per_store, then we will set
+    // default value to the flush_thread_num_per_store based on the disk num and core num
+    if (!props.contain("flush_thread_num_per_store")) {
+        size_t num_cores = std::thread::hardware_concurrency();
+        size_t disk_store_num = 1;
+
+        // get disk number
+        std::vector<starrocks::StorePath> paths;
+        auto status = starrocks::parse_conf_store_paths(starrocks::config::storage_root_path, &paths);
+        if (status.ok()) {
+            disk_store_num = std::max((size_t)1, paths.size());
+        }
+        // flush_thread_num_per_store * disk_store_num = num_cores / 2
+        size_t flush_thread_num_per_store = num_cores / 2 / disk_store_num;
```

Comment:
If the flush_thread_num_per_store is too much, the query load will be influenced much.

Reply:
If just want to benchmark, you can set by configuration.